components/virtual-node/virtual-node-write-service-impl/src/main/kotlin/net/corda/virtualnode/write/db/impl/writer/VirtualNodeDb.kt

```
@@ -77,4 +79,15 @@ class VirtualNodeDb(
             }
         }
     }
+
+    fun runCpiMigrations(dbChange: DbChange) {
+        dbConnections[DDL]?.let { dbConnection ->
```

I think we need to log an error and throw when the connection is null rather than silently fail.